File: src/test/java/javabot/web/views/ChangesViewTest.java
package javabot.web.views;

import io.dropwizard.views.freemarker.FreemarkerViewRenderer;
import javabot.dao.ChangeDao;
import javabot.model.Change;
import net.htmlparser.jericho.Source;
import org.testng.annotations.Test;

import javax.inject.Inject;
import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.time.LocalDateTime;
import java.util.Locale;

public class ChangesViewTest extends ViewsTest {
    @Inject
    private ChangeDao changeDao;

    @Test
    public void changes() throws IOException {
        createChanges(30);

        FreemarkerViewRenderer renderer = new FreemarkerViewRenderer();
        ByteArrayOutputStream output = new ByteArrayOutputStream();
        renderer.render(new ChangesView(getInjector(), new MockServletRequest(false), 0, new Change()), Locale.getDefault(), output);
        Source source = new Source(new ByteArrayInputStream(output.toByteArray()));

        previousDisabled(source);
        nextDisabled(source);
        checkRange(source, 1, 30, 30);

        output = new ByteArrayOutputStream();
        renderer.render(new ChangesView(getInjector(), new MockServletRequest(false), 0, new Change("change 2")), Locale.getDefault(), output);
        source = new Source(new ByteArrayInputStream(output.toByteArray()));

        previousDisabled(source);
        nextDisabled(source);
        checkRange(source, 1, 11, 11);
    }

    private void createChanges(final int count) {
        changeDao.deleteAll();
        for (int i = 0; i < count; i++) {
            Change change = new Change("change " + i);
            change.setChangeDate(LocalDateTime.now());
            changeDao.save(change);
        }

    }
}

File: src/test/kotlin/javabot/web/views/ChangesViewTest.kt
package javabot.web.views

import java.io.ByteArrayInputStream
import java.io.ByteArrayOutputStream
import java.time.LocalDateTime
import javabot.model.Change
import net.htmlparser.jericho.Source
import org.testng.annotations.Test

class ChangesViewTest : ViewsTest() {
    @Test
    fun changes() {
        createChanges(30)

        var output = ByteArrayOutputStream()
        var templateInstance =
            templateService.createChangesView(MockServletRequest(false), 0, null, null)
        var html = templateInstance.render()
        output.write(html.toByteArray())
        var source = Source(ByteArrayInputStream(output.toByteArray()))

        previousDisabled(source)
        nextDisabled(source)
        checkRange(source, 1, 30, 30)

        output = ByteArrayOutputStream()
        templateInstance =
            templateService.createChangesView(MockServletRequest(false), 0, "change 2", null)
        html = templateInstance.render()
        output.write(html.toByteArray())
        source = Source(ByteArrayInputStream(output.toByteArray()))

        previousDisabled(source)
        nextDisabled(source)
        checkRange(source, 1, 11, 11)
    }

    private fun createChanges(count: Int) {
        changeDao.deleteAll()
        for (i in 0..count - 1) {
            val change = Change("change " + i)
            change.changeDate = LocalDateTime.now()
            changeDao.save(change)
        }
    }
}
